src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java

```diff
@@ -679,40 +672,31 @@ private static int downloadBlobs(CloudBlobContainer container,
      */
     private static void downloadBlob(CloudBlob blob, FilePath downloadDir, boolean flattenDirectories,
 	    TaskListener listener) throws WAStorageException {
-	OutputStream fos = null;
-	try {
-	    FilePath downloadFile = new FilePath(downloadDir, blob.getName());
-
-	    // That filepath will contain all the directories and explicit virtual
-	    // paths, so if the user wanted it flattened, grab just the file name and
-	    // recreate the file path
-	    if (flattenDirectories) {
-		downloadFile = new FilePath(downloadDir, downloadFile.getName());
-	    }
+		try {
+			FilePath downloadFile = new FilePath(downloadDir, blob.getName());
+
+			// That filepath will contain all the directories and explicit virtual
+			// paths, so if the user wanted it flattened, grab just the file name and
+			// recreate the file path
+			if (flattenDirectories) {
+				downloadFile = new FilePath(downloadDir, downloadFile.getName());
+			}
 
-	    fos = downloadFile.write();
+			try (OutputStream fos = downloadFile.write()) {
+				long startTime = System.currentTimeMillis();
 
-	    long startTime = System.currentTimeMillis();
-            
-            blob.download(fos, null, getBlobRequestOptions(), Utils.updateUserAgent());
+				blob.download(fos, null, getBlobRequestOptions(), Utils.updateUserAgent());
```

Comment:
nit: Would be cleaner to wrap _just_ this blob.download() call in the try block for the stream